client/src/pages/DevelopmentMode.tsx

```diff
@@ -1,69 +1,51 @@
-import { useEffect, useCallback } from 'react';
+import { useEffect, useCallback, useState } from 'react';
 import { socket } from 'utils/socket';
 import styles from 'styles/DevelopmentMode.module.css';
 import globalStyles from 'styles/global.module.css';
 import classNames from 'classnames';
 import { Tabs, Tab } from 'components/Tabs';
 import VisualizerMain from './src/VisualizerMain';
+import { BackendState, CType } from './src/visualizer-component/types/backendType';
 
 const DevelopmentMode = () => {
-  const onGetBreakpoints = useCallback((data: any) => {
-    console.log(`Received message!!: ${data}`);
-  }, []);
-
-  useEffect(() => {
-    const onConnect = () => {
-      console.log('Connected!');
-      console.log('Emitting message to server...');
-      socket.emit('echo', 'example');
-      //      socket.emit("getBreakpoints", "121", "list2")
-      //      socket.emit("getBreakpoints", "122", "list2")
-      //      socket.emit("getBreakpoints", "123", "list2")
-      //      socket.emit("getBreakpoints", "124", "list2")
-      //      socket.emit("getBreakpoints", "125", "list2")
-      //      socket.emit("getBreakpoints", "126", "list2")
-
-      socket.emit('sendDummyData', '100');
-      socket.emit('sendDummyData', '101');
-      socket.emit('sendDummyData', '102');
-      socket.emit('sendDummyData', '103');
-      socket.emit('sendDummyData', '104');
-      socket.emit('sendDummyData', '105');
-    };
+  const [backendState, setBackendState] = useState<BackendState>({
+    '0x1': {
+      addr: '0x1',
+      type: CType.SINGLE_LINED_LIST_NODE,
+      is_pointer: false,
+      data: {
+        value: '27',
+        next: '0x0',
+      },
+    },
+  });
 
-    const onDisconnect = () => {
-      console.log('Disconnected!');
-    };
+  const [count, setCountState] = useState(100);
+  const onSendDummyData = useCallback((data: any) => {
```

Comment:
No need to cache this function with `useCallback()`, no dependencies

Reply:
Also lets add types for our socketio data! https://socket.io/docs/v4/typescript/